qiskit_nature/circuit/library/ansatzes/uvcc.py

```diff
@@ -152,21 +168,30 @@ def operators(self):  # pylint: disable=invalid-overridden-method
             # they will be left as None to be built at some later time.
             if self._check_uvcc_configuration(raise_on_failure=False):
                 # The qubit operators are cached by `EvolvedOperatorAnsatz` class. We only generate
-                # them from the `SecondQuantizedOp`s produced by the generators, if they're not already
-                # present.
+                # them from the `SecondQuantizedOp`s produced by the generators, if they're not
+                # already present. This behavior also enables the adaptive usage of the `UVCC` class
+                # by algorithms such as `AdaptVQE`.
                 excitation_ops = self.excitation_ops()
 
                 logger.debug("Converting SecondQuantizedOps into PauliSumOps...")
                 # Convert operators according to saved state in converter from the conversion of the
                 # main operator since these need to be compatible. If Z2 Symmetry tapering was done
-                # it may be that one or more excitation operators do not commute with the
-                # symmetry. Normally the converted operators are maintained at the same index by
-                # the converter inserting None as the result if an operator did not commute. Here
-                # we are not interested in that just getting the valid set of operators so that
-                # behavior is suppressed.
-                self.operators = self.qubit_converter.convert_match(
-                    excitation_ops, suppress_none=True
-                )
+                # it may be that one or more excitation operators do not commute with the symmetry.
+                # The converted operators are maintained at the same index by the converter inserting
+                # None as the result if an operator did not commute.
+                operators = self.qubit_converter.convert_match(excitation_ops, suppress_none=False)
+
+                # Here we are only interested in getting the valid set of operators. To ensure that
```

Comment:
By 'valid set` this means non-None. You switched from the conversion code dropping non-commuting ones to having None instead in order that you can remove them manually since you need to ensure any initial point to matches with the given terms and this is done based on their index order in the list which dropping them as before does not allow you too track.

Reply:
I updated the comment a bit to try to clarify what I'm doing here. Let me know if it still needs work.